docker-compose.yaml

```diff
@@ -11,16 +9,19 @@ services:
         container_name: ankerctl
         restart: unless-stopped
         build: .
+
+        # host-mode networking is required for pppp communication with the
+        # printer, since it is an asymmetrical udp protocol.
+        network_mode: host
```

Comment:
I'm finding that on mac, that network_mode: host does not work. The connection is refused. 

```
# Steps to purge all volumes and containers
docker system prune -f
docker volume rm ankermake-m5-protocol_ankerctl_vol
docker rmi $(docker image list -q ankerctl/ankerctl)
```

I've also tried setting the more permissive (but insecure) seeings

`FLASK_HOST=0.0.0.0`
```
ports:
  - 4470:4470
```

Sadly docker desktop for mac is quite limited, yet it is the most popular docker engine (rancher being probably the second most pouplar)

From the docker documentation

> The host networking driver only works on Linux hosts, and is not supported on Docker Desktop for Mac, Docker Desktop for Windows, or Docker EE for Windows Server.

https://docs.docker.com/network/host/

## Workaround

On mac, the following works instead of `network_mode: host`, however I have no idea if that will work on linux. 

```
    extra_hosts:
      - "host.docker.internal:host-gateway"
      ```

Reply:
Does this work with video streaming?

Almost any network setup will work with the http traffic for flask, but the problem is the crazy udp traffic ankerctl has to send to the printer.

So unless we have verified with video streaming (or, uploading a print job), we don't know if it works.

Reply:
My printer is torn apart at the moment so I'm not positive. 

Though after more research, it looks like the `extra_hosts` workaround is docker-desktop (mac) specific, so I don't think we should go with it. 

:shakes_fist: For being the standard for making code portable, docker sure isn't very portable is it? 

This is the closest workaround for linux (but doesn't work for mac) :shakes_fist: :shakes_fist:

```
    ports:
      - 127.0.0.1:4470:4470
    environment:
      - HOST_IP=<host-ip-address>
      ```

Reply:
Anything that relates to port 4470 is only for the http interface. UDP runs on port 32108, plus any number of randomly-opened ports (and that's part of the problem).

As far as I can tell, *any* docker network mode that involves NAT is going to break ankerctl's chance of talking to the printer.

So far, we have verified `host` and `macvlan` as working, and `default` (NAT-based) as broken. The rest are unknown.

Reply:
We'll do more research on this after v1 release :)